php-checks/src/main/java/org/sonar/php/checks/OpeningPHPTagCheck.java

```diff
@@ -19,38 +19,37 @@
  */
 package org.sonar.php.checks;
 
-import com.sonar.sslr.api.AstNode;
 import org.sonar.api.server.rule.RulesDefinition;
 import org.sonar.check.Priority;
 import org.sonar.check.Rule;
-import org.sonar.php.parser.PHPTokenType;
+import org.sonar.plugins.php.api.tree.ScriptTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
 import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
 import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
-import org.sonar.squidbridge.checks.SquidCheck;
-import org.sonar.sslr.parser.LexerlessGrammar;
 
 @Rule(
-  key = "S1757",
+  key = OpeningPHPTagCheck.KEY,
   name = "\"<?php\" and \"<?=\" tags should be used",
   priority = Priority.MINOR,
   tags = {Tags.CONVENTION, Tags.PSR1})
 @SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.READABILITY)
 @SqaleConstantRemediation("2min")
-public class OpeningPHPTagCheck extends SquidCheck<LexerlessGrammar> {
+public class OpeningPHPTagCheck extends PHPVisitorCheck {
+
+  public static final String KEY = "S1757";
+  private static final String MESSAGE = "Change this opening tag to either \"<?php\" or \"<?=\".";
 
   private static final String LONG_TAG = "<?php";
   private static final String SHORT_ECHO_TAG = "<?=";
   private static final String SHORT_TAG = "<?";
 
-  @Override
-  public void init() {
-    subscribeTo(PHPTokenType.FILE_OPENING_TAG);
-  }
 
   @Override
-  public void visitNode(AstNode astNode) {
-    if (isNotAuthorisedTag(astNode)) {
-      getContext().createLineViolation(this, "Change this opening tag to either \"<?php\" or \"<?=\".", getLineToReport(astNode));
+  public void visitScript(ScriptTree tree) {
+    String openingTag = tree.fileOpeningTagToken().text();
+
+    if (isNotAuthorisedTag(openingTag)) {
+      context().newIssue(this.KEY, MESSAGE).line(getLineToReport(openingTag));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2209)